ruby/bin/annotate

```diff
@@ -12,7 +12,13 @@ abort("#{junits_dir} does not exist") unless Dir.exist?(junits_dir)
 job_pattern = ENV['BUILDKITE_PLUGIN_JUNIT_ANNOTATE_JOB_UUID_FILE_PATTERN']
 job_pattern = '-(.*).xml' if !job_pattern || job_pattern.empty?
 
-class Failure < Struct.new(:name, :classname, :body, :job, :type)
+failure_format = ENV['BUILDKITE_PLUGIN_JUNIT_ANNOTATE_FAILURE_FORMAT']
+failure_format = 'classname' if !failure_format || failure_format.empty?
+if !%w(classname file).include?(failure_format)
+  raise ArgumentError.new("Invalid failure-format #{failure_format} provided -- leave this setting blank or specify `file` or `classname`")
+end
```

Comment:
@toolmantim I also added some incorrect arg handling here. I explored using enum as I saw [used in a separate plugin](https://github.com/bugcrowd/test-summary-buildkite-plugin/blob/master/plugin.yml#L19-L22) but when testing, it seemed to allow the incorrect arg to be used, resulting in summary text like:  `MyClass::Undertest failure text in ` with a blank filepath/classname. Lmk if you think this is overkill or if you prefer this to be handled another way, and I'll make those edits :)

Reply:
Yeah, we currently don't validate the args at runtime using the plugin.yml — but it's being added in buildkite/agent#748

Probably fine to leave in, and we can remove later. Only thing is I'd swap it out to use `abort` to make it consistent with the error handling at the top?